app/views/catalog/_show_sidebar.html.erb

```diff
@@ -12,5 +12,46 @@
     <% end %>
     <% file_counts = @document.file_counts.map { |group| "#{group[:extension]}(#{group[:file_count]})" } %>
     <%= render_sidebar_row "File Types: ", file_counts.join(", ") %>
+    <%= render_sidebar_doi_row @document.doi_url, @document.doi_value %>
   </table>
 </div>
+
+<script>
+  $(function() {
+    var setupCopyDoiToClipboard = function(linkId, textId) {
+
+      $("#copy-doi").click(function(x) {
+        var doi = this.dataset["url"];
+        // Copy value to the clipboard...
+        navigator.clipboard.writeText(doi).then(function() {
```

Comment:
This is all it's needed to copy a value to the clipboard. The rest of the JavaScript in this function is to toggle the display from "COPY" to "COPIED" and the little icon to have a checkbox to give an indication to the user that the value has indeed been copied to the clipboard.